doc/api/crypto.md

```diff
@@ -1041,6 +1041,26 @@ console.log(hash.digest('hex'));
 //   6a2da20943931e9834fc12cfe5bb47bbd9ae43489a30726962b576f4e3993e50
 ```
 
+### hash.copy([options])
+<!-- YAML
+added:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/29910
+-->
+
+* `options` {Object} [`stream.transform` options][]
+* Returns: {Hash}
+
+Creates a new `Hash` object that contains a deep copy of the internal state
+of the current `Hash` object.
+
+The optional `options` argument controls stream behavior. For XOF hash
+functions such as `'shake256'`, the `outputLength` option can be used to
+specify the desired output length in bytes.
+
+An error is thrown when an attempt is made to copy the `Hash` object after
+its [`hash.digest()`][] method has been called.
```

Comment:
An example here that shows how and why it's useful would be good. Perhaps an example of calculating a rolling hash given the original request?

Reply:
How about adding the following?

---

`hash.copy()` can be used to calculate rolling hashes of data at various points in the data stream.

```js
const hash = crypto.createHash('md5');

hash.update('abc');
const md1 = hash.copy().digest('hex'); // MD5 hash of 'abc'

hash.update('def');
const md2 = hash.copy().digest('hex'); // MD5 hash of 'abcdef'

hash.update('ghi');
const md3 = hash.digest('hex'); // MD5 hash of 'abcdefghi'
// Hash was not copied before final `.digest()` so cannot be reused further
```